course-v2/layouts/partials/course_description.html

```diff
@@ -1,7 +1,8 @@
 {{ $courseData := .context.Site.Data.course }}
 {{ $shouldCollapseDescription := false }}
 {{ with $courseData.course_description }}
-  {{ $shouldCollapseDescription = gt (len .) 320 }}
+  {{ $plainDescription := plainify . }}
+  {{ $shouldCollapseDescription = gt (len $plainDescription) 320 }}
```

Comment:
I noticed that `plainify .` might not be working as intended in this context. When testing, I found that $courseData.course_description seems to contain markdown rather than HTML. To confirm,  I did length check of `course_description` before and after `plainify` and it was same for `6.00-fall-2008`. 

This is the other bit of code for reference:
```

  {{ $shouldCollapseDescription = gt (len $plainDescription) 320 }}
    {{- truncate 320 (.context.RenderString $courseData.course_description) -}}
```

The problem is with the way we truncate here.

While we check non-HTML version of string for length, later on we truncate the HTML-inclusive string.

For example.
course_description=`6.00 Intro to CS and Programming has been retired from OCW. You can access the archived course on [DSpace – MIT’s digital repository](https://dspace.mit.edu/handle/1721.1/150580). Please see the list of [introductory programming courses](https://ocw.mit.edu/collections/introductory-programming/) and other programming courses from recent years`
length= 350

Same string truncated at 320:
`6.00 Intro to CS and Programming has been retired from OCW. You can access the archived course on [DSpace – MIT’s digital repository](https://dspace.mit.edu/handle/1721.1/150580). Please see the list of [introductory programming courses](https://ocw.mit.edu/collections/introductory-programming/) and other programming …`

the HTML equivalent of the complete string:
`6.00 Intro to CS and Programming has been retired from OCW. You can access the archived course on <a href="https://dspace.mit.edu/handle/1721.1/150580" target="_blank" rel="noopener">DSpace – MIT’s digital repository</a>. Please see the list of&nbsp;<a href="https://ocw.mit.edu/collections/introductory-programming/" target="_blank" rel="noopener">introductory programming courses</a> and other programming courses from recent years`
Length= 434 characters

320 characters of this (HTML inclusive string) becomes inconsistent with the comparison we did earlier (with markdown inclusive text). Which is why, I'm still getting the description truncation issue with the course `6.00-fall-2008`.

I think a possible solution might be to adjust the truncation operation to consider the rendered string length, like this:
`{{-  (.context.RenderString ($course_description | truncate 320))  -}}
`

Please let me know if this approach makes sense or if you have other suggestions.